File: uniffi_bindgen/src/bindings/kotlin/templates/UInt16Helper.kt
internal object FfiConverterU16 {
    fun lift(v: Short): UShort {
        return v.toUShort()
    }

    fun read(buf: ByteBuffer): UShort {
        return lift(buf.getShort())
    }

    fun lower(v: UShort): Short {
        return v.toShort()
    }

    fun write(v: UShort, buf: RustBufferBuilder) {
        buf.putShort(v.toShort())
    }
}

File: uniffi_bindgen/src/bindings/dart/templates/UInt16Helper.dart
fileprivate struct FfiConverterUInt16: FfiConverterPrimitive {
    typealias FfiType = UInt16
    typealias DartType = UInt16

    public static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> UInt16 {
        return try lift(readInt(&buf))
    }

    public static func write(_ value: DartType, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}
